luda-editor/luda-editor-client/src/editor/mod.rs

```diff
@@ -112,6 +119,84 @@ impl namui::Entity for Editor {
+                    }));
+                    spawn_local({
+                        let path = path.clone();
+                        let socket = self.socket.clone();
+                        async move {
+                            fn handle_error(path: String, started_at: Duration, error: String) {
+                                namui::log(format!("error on read_file: {:?}", error));
+                                namui::event::send(Box::new(
+                                    EditorEvent::SequenceLoadStateUpdateEvent {
+                                        path,
+                                        state: Some(SequenceLoadState {
+                                            started_at,
+                                            detail: SequenceLoadStateDetail::Failed { error },
+                                        }),
+                                    },
+                                ));
+                            }
+                            let result = socket
+                                .read_file(luda_editor_rpc::read_file::Request {
+                                    dest_path: path.clone(),
+                                })
+                                .await;
+                            match result {
+                                Ok(response) => {
+                                    let file = response.file;
+                                    match Sequence::try_from(file) {
+                                        Ok(sequence) => namui::event::send(Box::new(
+                                            EditorEvent::SequenceLoadStateUpdateEvent {
+                                                path: path.clone(),
+                                                state: Some(SequenceLoadState {
+                                                    started_at,
+                                                    detail: SequenceLoadStateDetail::Loaded {
+                                                        sequence,
+                                                    },
+                                                }),
+                                            },
+                                        )),
+                                        Err(error) => handle_error(path.clone(), started_at, error),
+                                    }
+                                }
+                                Err(error) => handle_error(path.clone(), started_at, error),
+                            }
+                        }
+                    });
+                }
+                EditorEvent::SequenceLoadStateUpdateEvent { path, state } => match state {
+                    Some(state) => {
+                        if let Some(old_state) = self.sequence_load_state_map.get(path) {
+                            if old_state.started_at > state.started_at {
+                                return;
+                            }
+                        }
```

Comment:
If `SequenceLoadStateUpdateEvent` is old event. Ignore update

Reply:
To achieve what you want, I guess sequence_load_state_map should be updated before sending rpc at here https://github.com/NamseEnt/namseent/pull/55/files#diff-01d4ae1db8edb49f41fba1d5f096b9f6ee34f66399f1eda95feab765eaaca687R147.
Let's think about this case,

Request A and Request B,
Request A started at 00:00
Request B started at 00:01
Request B finished at 00:02, nothing in map, so it updates map
Request A finished at 00:03, B is older than A, so it updates map.

It depends on the definition what is the 'old event'. Is it old if event fired later or rpc started later? If rpc-later is older, please update this logic like I said.

Reply:
Actually, status is updated before sending rpc.
And started_at means when Editor got `SequenceLoadEvent`. Regardless of when request was finished.

When Editor got `SequenceLoadEvent`. It fires `SequenceLoadStateUpdateEvent` immediately to set state to `Loading`.
And LoadState's started_at means this moment (When Editor got `SequenceLoadEvent`).
https://github.com/NamseEnt/namseent/pull/55/files#diff-01d4ae1db8edb49f41fba1d5f096b9f6ee34f66399f1eda95feab765eaaca687R123-R130

And then send rpc, update result
https://github.com/NamseEnt/namseent/pull/55/files#diff-01d4ae1db8edb49f41fba1d5f096b9f6ee34f66399f1eda95feab765eaaca687R131-R174

So your example works like as follows
Request A and Request B,
1. Request A started at 00:00. **Update state to `Loading:started at 00:00`**
2. Request B started at 00:01. **Update state to `Loading:started at 00:01`** because Request B is newer
3. Request B finished at 00:02, There is `Loading:started at 00:01`. Request B is not older than `Loading:started at 00:01`. Update to `Loaded:started at 00:01`.
4. Request A finished at 00:03, B is **newer** than A, so update(`Loaded:started at 00:00`) **will be ignored**.